File: examples/governance_static_go/domain/reject_assessment.go
package domain

import (
	"time"
	"fmt"
)

type RejectAssessment struct {
	AssessmentId string `json:"assessment_id"`
}

func (c RejectAssessment) CommandName() string { return "RejectAssessment" }

func (c RejectAssessment) Execute(repo AssessmentRepository) (*Assessment, *RejectedAssessment, error) {
	existing, err := repo.Find(c.AssessmentId)
	if err != nil {
		return nil, nil, err
	}
	if existing == nil {
		return nil, nil, fmt.Errorf("Assessment not found: %s", c.AssessmentId)
	}
	if existing.Status != "pending" && existing.Status != "submitted" {
		return nil, nil, fmt.Errorf("cannot RejectAssessment: Assessment is in %s state", existing.Status)
	}
	existing.Status = "rejected"
	existing.UpdatedAt = time.Now()
	if err := existing.Validate(); err != nil {
		return nil, nil, err
	}
	if err := repo.Save(existing); err != nil {
		return nil, nil, err
	}
	event := RejectedAssessment{
		AggregateID: existing.ID,
		AssessmentId: c.AssessmentId,
		OccurredAt: time.Now(),
	}
	return existing, &event, nil
}

File: examples/governance/risk_assessment_domain/lib/risk_assessment_domain/assessment/commands/reject_assessment.rb
module RiskAssessmentDomain
  class Assessment
    module Commands
      class RejectAssessment
        include Hecks::Command
        emits "RejectedAssessment"

        attr_reader :assessment_id

        def initialize(assessment_id: nil)
          @assessment_id = assessment_id
        end

        def call
          existing = repository.find(assessment_id)
          if existing
            unless ["pending", "submitted"].include?(existing.status)
              raise RiskAssessmentDomain::Error, "Cannot RejectAssessment: status must be one of pending, submitted, got '#{existing.status}'"
            end
            Assessment.new(
              id: existing.id,
              model_id: existing.model_id,
              assessor_id: existing.assessor_id,
              risk_level: existing.risk_level,
              bias_score: existing.bias_score,
              safety_score: existing.safety_score,
              transparency_score: existing.transparency_score,
              overall_score: existing.overall_score,
              submitted_at: existing.submitted_at,
              findings: existing.findings,
              mitigations: existing.mitigations,
              status: "rejected"
            )
          else
            raise RiskAssessmentDomain::Error, "Assessment not found: #{assessment_id}"
          end
        end
      end
    end
  end
end
